src/dataconnect/schemaMigration.ts

```diff
@@ -62,6 +63,23 @@ export async function migrateSchema(args: {
     databaseId,
     /* linkIfNotConnected=*/ true,
   );
+
+  if (experiments.isEnabled("compatiblemode")) {
+    if (schema.primaryDatasource.postgresql?.schemaValidation) {
```

Comment:
Ah, I see that `getIdentifiers()` gets called. By the time you hit this block, it's ensured that postgresql is defined. You can probably just simplify to:

```
if (experiments.isEnabled("compatiblemode")) {
   schema.primaryDatasource.postgresql.schemaValidation = "COMPATIBLE";
}
```

Reply:
It still gives an error without the if statements